### PR TITLE
whirlpool-gui: remove

### PR DESCRIPTION
**Pull request by brandonweeks**
###### Motivation for this change

whirlpool-gui depends on the very out dated Electron 7. Upstream does
not appear to be tracking newer versions of Electron.

https://code.samourai.io/whirlpool/whirlpool-gui

Attempts to migrate to newer versions of Electron with the latest upstream code were unsuccessful.

This change also removes the (now unused) outdated Electron versions.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1113,6 +1115,7 @@ mapAliases ({
   webkitgtk24x-gtk3 = throw "webkitgtk24x-gtk3 has been removed because it's insecure. Please use webkitgtk."; # added 2019-12-05
   webkitgtk24x-gtk2 = throw "webkitgtk24x-gtk2 has been removed because it's insecure. Please use webkitgtk."; # added 2019-12-05
   weechat-matrix-bridge = weechatScripts.weechat-matrix-bridge; # added 2018-09-06
+  whirlpool-gui = "throw whirlpool-gui has been removed nixpkgs as it depended on an insecure version of Electron."; # added 2022-02-08q
```
Comment (wamserma) [inline pkgs/top-level/aliases.nix:1118]:
```suggestion
  whirlpool-gui = throw "whirlpool-gui has been removed as it depended on an insecure version of Electron."; # added 2022-02-08
```

**Review by wamserma — CHANGES_REQUESTED** [review]

**Review by wamserma — APPROVED** [review]

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -240,10 +240,12 @@ mapAliases ({
   ec2_api_tools = ec2-api-tools; # added 2021-10-08
   elasticmq = throw "elasticmq has been removed in favour of elasticmq-server-bin"; # added 2021-01-17
   elasticsearch7-oss = throw "elasticsearch7-oss has been removed, as the distribution is no longer provided by upstream. https://github.com/NixOS/nixpkgs/pull/114456"; # added 2021-06-09
-  electron_3 = throw "electron_3 has been removed in favor of newer versions"; # added 2021-01-06
-  electron_4 = throw "electron_4 has been removed in favor of newer versions"; # added 2021-01-06
-  electron_5 = throw "electron_5 has been removed in favor of newer versions"; # added 2021-01-06
-  electron_6 = throw "electron_6 has been removed in favor of newer versions"; # added 2021-01-06
+  electron_3 = throw "electron_3 has been removed in favor of newer versions"; # added 2022-01-06
```
Comment (c0bw3b) [inline pkgs/top-level/aliases.nix:243]: Why change the date of the aliases addition for electron_3 to _6 ?

Comment (brandonweeks) [reply pkgs/top-level/aliases.nix:243]: Fixing my own typo from 6730766. 2020 to 2022 has just blended together.

Comment (c0bw3b) [reply pkgs/top-level/aliases.nix:243]: Oh ok

**Comment by c0bw3b** [comment]
Could you rebase on current master and resolve the conflict on `aliases.nix`?

**Comment by brandonweeks** [comment]
Done.

**Comment by brandonweeks** [comment]
> Thanks. Aren't you missing the alias for `whirlpool-gui` itself?
> 
> ```nix
> whirlpool-gui = throw "whirlpool-gui has been removed as it depended on an insecure version of Electron."; # added 2022-02-08
> ```

Indeed I am, thank you, must have got dropped in the rebase. Sorry about that.

**Review by c0bw3b — APPROVED** [review]
Just to recheck eval

Result of `nixpkgs-review pr 158581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Pull request merged**